pkg/machines/components/createDialog.jsx

```diff
@@ -0,0 +1,376 @@
+                        <td>
+                            <input id="start-vm" className="form-control start-vm-checkbox" type="checkbox"
+                                   checked={this.props.vmParams.startVm}
+                                   onChange={this.onChangedEventChecked.bind(this, 'startVm')}/>
+                        </td>
+                    </tr>
+                </table>
+            </div>
+        );
+    }
+}
+
+CreateVM.propTypes = {
+    valuesChanged: PropTypes.func.isRequired,
+    initialData: PropTypes.string.isRequired,
+    vendorMap: PropTypes.object.isRequired,
+    vendorNames: PropTypes.array.isRequired,
+};
+
+function validateParams (vmParams) {
+    if (isEmpty(vmParams.vmName)) {
+        return _("Name cannot be empty");
+    }
+
+    vmParams.vmName = vmParams.vmName.trim();
+    if (isEmpty(vmParams.vmName)) {
+        return _("Name cannot be empty characters only");
+    }
+
+
+    if (!isEmpty(vmParams.source)) {
+        vmParams.source = vmParams.source.trim();
+        if (!vmParams.source.startsWith("http") &&
+            !vmParams.source.startsWith("ftp") &&
+            !vmParams.source.startsWith("nfs")) {
+            return _("Source should start with http, ftp or nfs protocol");
+        }
+    }
+
+    if (vmParams.memorySize <= 0) {
+        return _("Memory must be positive number");
+    }
+
+    if (vmParams.storageSize < 0) {
+        return _("Storage Size should not be negative number");
+    }
+}
+
+const createDialog = (dispatch, osInfoList) => {
+    const vmParams = {
+        'vmName': null,
+        'source': null,
+        'vendor': NOT_SPECIFIED,
+        "os": OTHER_OS_SHORT_ID,
+        'memorySize': 0, // MiB
+        'storageSize': 0, // GiB
+        'startVm': false,
+    };
+
+    const changeParams = (key, value) => {
+        vmParams[key] = value;
+    };
+
+    const vendors = prepareVendors(osInfoList);
+
+    const dialogBody = (
+        <CreateVM vmParams={vmParams}
+                  familyList={vendors.familyList}
+                  familyMap={vendors.familyMap}
+                  vendorMap={vendors.vendorMap}
+                  valuesChanged={changeParams}/>
+    );
+
+    const dialogProps = {
+        'title': _("Create New Virtual Machine"),
+        'body': dialogBody,
+    };
+
+    // also test modifying properties in subsequent render calls
+    const footerProps = {
+        'actions': [
+            {
+                'clicked': () => {
+                    const error = validateParams(vmParams)
+                    if (error) {
+                        return cockpit.defer().reject(error).promise;
+                    } else {
+                        return dispatch(createVm(vmParams));
+                    }
+                },
+                'caption': _("Create"),
+                'style': 'primary',
+            },
+        ],
+    };
+
+    DialogPattern.show_modal_dialog(dialogProps, footerProps);
+};
+
+export default createDialog;
```

Comment:
`createDialog` is misleading for newcomers. Please consider renaming the file/function to `createVmDialog`.